types/mongodb/test/cursor.ts

```diff
@@ -34,11 +34,37 @@ async function run() {
     .stream();
 
   collection.find().project({});
-  collection.find().project({notExistingField: 1});
-  collection.find().sort({'fruitTags.name': -1, numberField: -1, notExistingField: -1});
+  collection.find().project({notExistingField: 1, });
   collection.find().sort({text: {$meta: 'textScore'}, notExistingField: -1});
   collection.find().sort({});
 
+  interface TypedDoc {
+    name: string;
+    age: number;
+    tag: {
+      name: string;
+    };
+  }
+  const typedCollection = db.collection<TypedDoc>('test');
+  typedCollection.find({name: 'name'}, {}).map(x => x.tag);
+  typedCollection.find({'tag.name': 'name'}, {});
+  typedCollection.find({'tag.name': 'name'}, {projection: {name: 1, max: {$max: []}}})
+    .sort({score: {$meta: 'textScore'}})
+    .map(x => x.tag);
```

Comment:
Might also need to add a test for nested fields in projection:
```suggestion
  typedCollection.find({'tag.name': 'name'}, {projection: {'tag.name': 1, max: {$max: []}}})
    .sort({score: {$meta: 'textScore'}})
    .map(x => x.tag);
```